src/c++/perf_analyzer/infer_context.cc

```diff
@@ -206,15 +206,15 @@ InferContext::ValidateOutputs(const cb::InferResult* result_ptr)
       size_t byte_size = 0;
       result_ptr->RawData(infer_data_.outputs_[i]->Name(), &buf, &byte_size);
       for (const auto& expected : infer_data_.expected_outputs_[i]) {
-        if (byte_size < expected.second) {
+        if (byte_size < expected.batch1_size) {
```

Comment:
Should this whole thing be predicated on a check on the valid bit?

Reply:
i always shy away from asserts because they cause an abrupt failure. That said, if the whole thing is gated by the valid bit check and somehow invalid data gets into the vector, you would basically skip this section and other code would abruptly fail.
i guess use an assert with good messaging for debugging.

Reply:
My concern is that there are not any other asserts in the codebase. I could just make it explicitly fail if it is invalid

Reply:
thats fine with me too.

Reply:
done